src/convertToObject.js

```diff
@@ -3,16 +3,35 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example
+ * in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const separateStylesInArray = sourceString
+    .split(';').map(oneStyle => oneStyle.trim())
+    .filter(oneStyle => oneStyle !== '');
```

Comment:
I guess you can avoid using `map` and `filter` if you are using `reduce` later. You can implement this logic in `reduce` to avoid using so many loops